Compiler.cmake

```diff
@@ -11,6 +11,7 @@
 # Input Variables
 # * COMMON_MINIMUM_GCC_VERSION check for a minimum gcc version, default 4.4
 # * COMMON_USE_CXX03 When set, do not enable C++11 language features
+# * COMMON_DISABLE When set, do not set any flags
```

Comment:
@eile 
w/ cmake 3, there is https://cmake.org/cmake/help/v3.0/module/CheckCXXCompilerFlag.html for checking flaga compatibility.  I haven't looked if this exists in the cmake 2.x series.

I will probably come down sometime so we can discuss this face-to-face.  @juanchopanza let me know if you interested in being there.